pkg/resource/template/resource.go

```diff
@@ -155,16 +155,31 @@ func (t *TemplateResource) createStageFile() error {
 	}
 
 	if err = tmpl.Execute(temp, nil); err != nil {
-		temp.Close()
-		os.Remove(temp.Name())
+		// The key error to return is the failure to execute.
+		// to preserve that error ignore the errors in close and clean
+		temp.Close()           // nolint:errcheck
+		os.Remove(temp.Name()) // nolint:errcheck
 		return err
 	}
-	defer temp.Close()
+	defer func() {
+		if e := temp.Close(); e != nil {
+			if ferr == nil {
+				ferr = e
+			}
+		}
+		if ferr != nil {
+			log.Error(ferr.Error())
```

Comment:
Feels like we should be logging the errors closer to the top of the call stack, unless we're swallowing an error and think it would be useful to know about it.